test/sqllogictest/arrays.slt

```diff
@@ -852,3 +852,64 @@ SELECT ARRAY[ARRAY[customer.first_name], ARRAY[customer.zip], ARRAY[customer.id:
 query error db error: ERROR: array_in not yet supported
 SELECT array_in('', 0, 0)
 ----
+
+# Array casting for column expressions
+
+statement ok
+CREATE TABLE array_t2 (
+    a int[],
+    b string[][],
+    c varchar[],
+    d int[][]
+);
+
+statement ok
+INSERT INTO array_t2 VALUES (
+    array[1, 2, 3, 4, 5],
+    array[['t1', 't2', 't3']],
+    array['test1', 'test2', 'test3', 'test4']::varchar[],
+    array[[0, 1, 2], [3, 4, 5], [6, 7, 8]]
+);
+
+query T
+SELECT a::string[]::int[]::text[]::float8[] FROM array_t2;
+----
+{1,2,3,4,5}
+
+query T
+SELECT b::text[] FROM array_t2;
+----
+{{t1,t2,t3}}
+
+query T
+SELECT b::text[] FROM array_t2;
+----
+{{t1,t2,t3}}
```

Comment:
These look identical, unless I'm missing something.

Reply:
🤦🏾 Thanks for catching that!